web/app/Common/Asset.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Wikijump\Common;
+
+use Exception;
+use Illuminate\Support\Facades\App;
+use Illuminate\Support\Str;
+use Innocenzi\Vite\ManifestEntry;
+use Innocenzi\Vite\Vite;
+
+/**
+ * Utility for resolving and working with assets by their unresolved path.
+ */
+class Asset
+{
+    /** Name of the asset. */
+    private string $name;
+
+    /** Unresolved path to the asset. */
+    private string $path;
+
+    /** Resolved path to the asset. */
+    private ?string $resolved_path;
+
+    /** Entry for the asset in the Vite manifest. */
+    private ?ManifestEntry $manifest_entry;
+
+    /** Path added to entrypoints not given with an explicit path. */
+    private const ENTRYPOINT_FOLDER = 'resources/scripts/';
+
+    /**
+     * @param string $name Name or path of the asset.
+     */
+    public function __construct(string $name)
+    {
+        $this->name = $name;
+
+        // if the path doesn't start with a `/`, it's a Vite path
+        if (!Str::startsWith($name, '/')) {
+            $vite = App::make(Vite::class);
+
+            // check if path has any slashes in it
+            $is_shorthand = strpos($name, '/') === false;
```

Comment:
> `int|false`

:frowning_face:

Reply:
?
`strpos` returns false if the substring can't be found. Otherwise, the `int` returned is the index. Unless this is a joke I'm not getting lmao

Reply:
No just displeasure at weird PHP typing lol